daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala

```diff
@@ -378,14 +392,47 @@ object Value extends CidContainer1WithDefaultCidResolver[Value] {
       implicit val `V0 Order`: Order[V0] = Order.fromScalaOrdering[String] contramap (_.coid)
     }
 
-    final case class V1(discriminator: crypto.Hash, suffix: Bytes = Bytes.Empty)
-        extends AbsoluteContractId {
+    final class V1 private(val discriminator: crypto.Hash, val suffix: Bytes) extends AbsoluteContractId {
```

Comment:
What's wrong with the `case class`?

Reply:
`copy` lets you break construction-enforced invariants. I think you can deal with this by defining a private `copy` instead, but 🤷‍♂️

Reply:
Cool. I did not know we can override the `copy` method.   

With the current settings we cannot make the overriding method private if we do not use it. 
@S11001001 What do you think of my solution. Do you think to any better way ?

Reply:
Your solution seems perfectly sensible to me. 👍